src/user-event/press/__tests__/press.test.tsx

```diff
@@ -1,67 +1,291 @@
+        <Text>press me</Text>
+      </TouchableHighlight>
+    );
+    await userEvent.press(screen.getByText('press me'));
+
+    expect(mockOnPress).toHaveBeenCalled();
+  });
+
+  test('works on Text', async () => {
+    const { events, logEvent } = createEventLogger();
+
+    render(
       <Text
-        testID="view"
         onPress={logEvent('press')}
         onPressIn={logEvent('pressIn')}
         onPressOut={logEvent('pressOut')}
+      >
+        press me
+      </Text>
+    );
+    await userEvent.press(screen.getByText('press me'));
+
+    expect(getEventsName(events)).toEqual(['pressIn', 'press', 'pressOut']);
+  });
+
+  test('doesnt trigger on disabled Text', async () => {
+    const { events, logEvent } = createEventLogger();
+
+    render(
+      <Text
+        onPress={logEvent('press')}
+        onPressIn={logEvent('pressIn')}
+        onPressOut={logEvent('pressOut')}
+        disabled
+      >
+        press me
+      </Text>
+    );
+    await userEvent.press(screen.getByText('press me'));
+
+    expect(events).toEqual([]);
+  });
+
+  test('doesnt trigger on Text with disabled pointer events', async () => {
+    const { events, logEvent } = createEventLogger();
+
+    render(
+      <View pointerEvents="box-only">
+        <Text
+          onPress={logEvent('press')}
+          onPressIn={logEvent('pressIn')}
+          onPressOut={logEvent('pressOut')}
+        >
+          press me
+        </Text>
+      </View>
+    );
+    await userEvent.press(screen.getByText('press me'));
+
+    expect(events).toEqual([]);
+  });
+
+  test('works on TetInput', async () => {
+    const { events, logEvent } = createEventLogger();
+
+    render(
+      <TextInput
+        placeholder="email"
+        onPressIn={logEvent('pressIn')}
+        onPressOut={logEvent('pressOut')}
       />
     );
+    await userEvent.press(screen.getByPlaceholderText('email'));
+
+    expect(getEventsName(events)).toEqual(['pressIn', 'pressOut']);
+  });
+
+  test('does not call onPressIn and onPressOut on non editable TetInput', async () => {
+    const { events, logEvent } = createEventLogger();
+
+    render(
+      <TextInput
+        placeholder="email"
+        editable={false}
+        onPressIn={logEvent('pressIn')}
+        onPressOut={logEvent('pressOut')}
+      />
+    );
+    await userEvent.press(screen.getByPlaceholderText('email'));
+    expect(events).toEqual([]);
+  });
+
+  test('does not call onPressIn and onPressOut on TetInput with pointer events disabled', async () => {
+    const { events, logEvent } = createEventLogger();
+
+    render(
+      <TextInput
+        placeholder="email"
+        pointerEvents="box-none"
+        onPressIn={logEvent('pressIn')}
+        onPressOut={logEvent('pressOut')}
+      />
+    );
+    await userEvent.press(screen.getByPlaceholderText('email'));
+    expect(events).toEqual([]);
+  });
+
+  test('press is accessible directly in userEvent', async () => {
```

Comment:
This test seems redundant.

Reply:
It is in a way yes, however it explicitly states that it tests that so I think it's still useful. It's possible that other tests are refactored to use the return of the `setup` function and this would not be tested anymore. I believe that every thing we want to test should be explicitly said in a test name, wdyt? If the intention behind every test is not explicit then it's very difficult to know which cases are covered and which aren't and I think having a bit of duplication is a small price to pay for that